lib/min_heap.rb

```diff
@@ -8,27 +8,29 @@ def initialize(key, value)
 end
 
 class MinHeap
-
   def initialize
     @store = []
   end
 
   # This method adds a HeapNode instance to the heap
-  # Time Complexity: ?
-  # Space Complexity: ?
+  # Time Complexity: O(log n) - where n is the number of heap nodes
+  # Space Complexity: O(1) - constant
```

Comment:
Since `heap_up` is recursive... You do have some space complexity here.  The system stack.